hazelcast/src/main/java/com/hazelcast/internal/serialization/impl/portable/FieldDefinitionImpl.java

```diff
@@ -27,9 +28,21 @@
     private final int classId;
     private final int factoryId;
     private final int version;
+    private final ClassDefinition classDefinition;
 
     public FieldDefinitionImpl(int index, String fieldName, FieldType type, int version) {
-        this(index, fieldName, type, 0, 0, version);
+        this(index, fieldName, type, 0, 0, version, null);
+    }
+
+    public FieldDefinitionImpl(int index, String fieldName, FieldType type, int factoryId,
```

Comment:
classDefinition already has factoryId classId and version, we don't need the same parameters